extensions/binding/org.eclipse.smarthome.binding.homematic/src/main/java/org/eclipse/smarthome/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -140,13 +153,25 @@ private void configureThingProperties() {
      * Schedules a reinitialization, if the Homematic gateway is not reachable at bridge startup.
      */
     private void scheduleReinitialize() {
-        scheduler.schedule(() -> {
-            initialize();
-        }, REINITIALIZE_DELAY_SECONDS, TimeUnit.SECONDS);
+        if (!isDisposed) {
+            initializeFuture = scheduler.schedule(this::initializeInternal, REINITIALIZE_DELAY_SECONDS,
+                    TimeUnit.SECONDS);
+        }
     }
 
     @Override
     public void dispose() {
+        synchronized (initDisposeLock) {
+            if (initializeFuture != null) {
+                initializeFuture.cancel(true);
+            }
+
+            disposeInternal();
+            isDisposed = true;
+        }
+    }
+
+    private void disposeInternal() {
         logger.debug("Disposing bridge '{}'", getThing().getUID().getId());
         super.dispose();
```

Comment:
is `super.dispose` really intended to be called here? When `initializeInternal` fails exceptionally `super.dispose` is called but then `scheduleReinitialize` will keep the handler up. I suggest to only clean up internal stuff here and call `super.dispose` on `this.dispose`.

Reply:
Good point, I moved the call to the `dispose()` method.